sparse_strips/vello_hybrid/shaders/sparse_strip_renderer.wgsl

```diff
@@ -0,0 +1,131 @@
+    let pix_y = f32(y0) + y * f32(config.strip_height);
+    // Convert pixel coordinates to normalized device coordinates (NDC)
+    // NDC ranges from -1 to 1, with (0,0) at the center of the viewport
+    let ndc_x = (pix_x + 0.5) * 2.0 / f32(config.width) - 1.0;
+    let ndc_y = 1.0 - (pix_y + 0.5) * 2.0 / f32(config.height);
+
+    out.position = vec4<f32>(ndc_x, ndc_y, 0.0, 1.0);
+    out.tex_coord = vec2<f32>(f32(instance.col) + x * f32(width), y * f32(config.strip_height));
+    out.color = instance.rgba;
+    return out;
+}
+
+@group(0) @binding(0)
+var alphas_texture: texture_2d<u32>;
+
+@fragment
+fn fs_main(in: VertexOutput) -> @location(0) vec4<f32> {
+    let x = u32(floor(in.tex_coord.x));
+    var alpha = 1.0;
+    // Determine if the current fragment is within the dense (alpha) region
+    // If so, sample the alpha value from the texture; otherwise, alpha remains fully opaque (1.0)
+    // TODO: This is a branch, but we can make it branchless by using a select
+    // would it be faster to do a texture lookup for every pixel?
+    if x < in.dense_end {
+        let y = u32(floor(in.tex_coord.y));
+        // Retrieve alpha value from the texture
+        // Calculate texture coordinates based on the fragment's x-position
+        // Since we store 4 alpha values per texel, divide x by 4 to get the texel position
+        let alphas_index = x;
+        let tex_dimensions = textureDimensions(alphas_texture);
+        let alphas_tex_width = tex_dimensions.x;
+        // Which texel contains our alpha value
+        let texel_index = alphas_index / 4u;
+        // Which channel (R,G,B,A) in the texel
+        let channel_index = alphas_index % 4u;
+        // Calculate texture coordinates using bitwise operations
+        // This is more efficient than using modulo and division when width is a power of 2
+        let alphas_tex_width_bits = firstTrailingBit(alphas_tex_width);
+        let tex_x = texel_index & (alphas_tex_width - 1u);
+        let tex_y = texel_index >> alphas_tex_width_bits;                  
+        
+        // Load all 4 channels from the texture
+        let rgba_values = textureLoad(alphas_texture, vec2<u32>(tex_x, tex_y), 0);
+        
+        // Get the alphas from the appropriate RGBA channel based on the index
+        let alphas_u32 = unpack_alphas_from_channel(rgba_values, channel_index);
+        // Extract the alpha value for the current y-position from the packed u32 data
+        alpha = f32((alphas_u32 >> (y * 8u)) & 0xffu) * (1.0 / 255.0);
+    }
+    // Apply the alpha value to the unpacked RGBA color
+    return alpha * unpack4x8unorm(in.color);
+}
+
+fn unpack_alphas_from_channel(rgba: vec4<u32>, channel_index: u32) -> u32 {
+    switch channel_index {
+        case 0u: { return rgba.x; }
+        case 1u: { return rgba.y; }
+        case 2u: { return rgba.z; }
+        case 3u: { return rgba.w; }
+        // Fallback, should never happen
+        default: { return rgba.x; }
+    }
+}
```

Comment:
This should have a final newline. You can normally configure your editor to add these automatically.

Unfortunately, I'm not aware of a good formatter for WGSL files, so this can't really be checked on CI.

Reply:
Oh, my bad! I missed that.

> You can normally configure your editor to add these automatically.

Yeah, I recently switched to vscode for some projects and haven’t configured everything yet.

Fixed in 93038ab